micrometer-observation/src/main/java/io/micrometer/observation/SimpleObservation.java

```diff
@@ -268,7 +277,25 @@ public void close() {
         @Override
         public void reset() {
             this.registry.setCurrentObservationScope(null);
-            this.currentObservation.notifyOnScopeReset();
+            Scope scope = this;
+            while (scope != null) {
+                scope.close();
+                SimpleScope simpleScope = (SimpleScope) scope;
+                scope = simpleScope.previousObservationScope;
+            }
+        }
+
+        @Override
+        public void makeCurrent() {
```

Comment:
I'd add a comment that this logic is justified only under the circumstance that before it is executed, a reset was called immediately before.